api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_stats.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+from concurrent import futures
+from typing import Any, Generator, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import tiledbsoma as soma
+
+from ..util._eager_iter import _EagerIterator
+from ._online import MeanAccumulator, MeanVarianceAccumulator
+
+
+def mean_variance(
+    query: soma.ExperimentAxisQuery,
+    layer: str = "raw",
+    axis: int = 0,
+    calculate_mean: bool = False,
+    calculate_variance: bool = False,
+    ddof: int = 1,
+) -> pd.DataFrame:
+    """
+    Calculate  mean and/or variance along the `obs` axis from query results. Calculations
+    are done in an accumulative chunked fashion.
```

Comment:
Let's clarify the denominator for our calculation, some users may expect the average of non-zero values. Which we should implement shortly after landing this PR.

```suggestion
    are done in an accumulative chunked fashion. For the mean and variance calculations, the total number of elements (N) is the corresponding dimension size: for column-wise calculations (`axis = 0`) N is number of rows, for row-wise calculations (`axis = 1`)  N is number of columns.
```